conn.go

```diff
@@ -24,25 +24,32 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+type Entity interface {
+	HandleResponse(netlink.Message)
+}
+
 // A Conn represents a netlink connection of the nftables family.
 //
 // All methods return their input, so that variables can be defined from string
 // literals when desired.
 //
 // Commands are buffered. Flush sends all buffered commands in a single batch.
 type Conn struct {
-	TestDial nltest.Func // for testing only; passed to nltest.Dial
-	NetNS    int         // Network namespace netlink will interact with.
 	sync.Mutex
-	messages []netlink.Message
-	err      error
+	TestDial   nltest.Func // for testing only; passed to nltest.Dial
+	NetNS      int         // Network namespace netlink will interact with.
+	entities   map[int]Entity
+	messagesMu sync.Mutex
```

Comment:
Move messagesMu and messages down and separate them with a newline (to form a group of fields):

```
type Conn struct {
  TestDial nltest.Func
  // …
  entities map[int]Entity

  messagesMu sync.Mutex
  messages []netlink.Message
}
```